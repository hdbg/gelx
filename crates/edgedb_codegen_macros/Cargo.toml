[package]
name = "edgedb_codegen_macros"
version.workspace = true
authors.workspace = true
edition.workspace = true
homepage.workspace = true
license.workspace = true
repository.workspace = true
rust-version.workspace = true

[lib]
proc-macro = true

[dependencies]
edgedb_codegen_core = { workspace = true }
proc-macro2 = { workspace = true }
quote = { workspace = true }
syn = { workspace = true }

[dev-dependencies]
trybuild = { workspace = true }
insta = { workspace = true, features = ["yaml", "ron"] }
assert2 = { workspace = true }
rstest = { workspace = true }
edgedb-derive = { workspace = true }
edgedb-errors = { workspace = true }
edgedb-protocol = { workspace = true, features = ["all-types", "with-serde"] }
edgedb-tokio = { workspace = true, features = ["unstable", "derive"] }
typed-builder = { workspace = true }
serde = { workspace = true, features = ["derive"] }
tokio = { workspace = true, features = ["time", "test-util"] }

[lints]
workspace = true

[features]
serde = []
query = []
builder = []
