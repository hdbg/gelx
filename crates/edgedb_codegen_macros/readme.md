# `edgedb_codegen_macros`

<br />

> This crate contains the raw macros for the `edgedb_codegen` crate.
