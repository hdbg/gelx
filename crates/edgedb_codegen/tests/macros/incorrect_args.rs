use edgedb_codegen_macros::edgedb_query;

fn main() {
	edgedb_query!(example_module, invalid);
}
