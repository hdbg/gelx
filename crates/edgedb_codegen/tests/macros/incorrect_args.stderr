error: expected string literal
 --> tests/macros/incorrect_args.rs:4:32
  |
4 |     edgedb_query!(example_module, invalid);
  |                                   ^^^^^^^
